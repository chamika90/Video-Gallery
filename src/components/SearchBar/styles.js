import {StyleSheet} from 'react-native';
import {theme} from '../../config/theme';

const {colors} = theme;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: colors.primarySearchBarBackgroundColor,
    alignItems: 'center',
    marginBottom: 5,
    borderRadius: 6,
  },
  input: {
    flex: 3,
    //backgroundColor: 'green',
  },
  eyeIcon: {
    flex: 0.5,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default styles;
