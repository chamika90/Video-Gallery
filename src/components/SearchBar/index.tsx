import React from 'react';
import {View, TextInput, TouchableOpacity} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome5';

import styles from './styles';
import {theme} from '../../config/theme';

const {colors} = theme;

interface Props {
  placeholder: string;
  onTextChange: (value: string) => void;
}

/*
 * Custom Input
 */
const SearchBar: React.FC<Props> = ({placeholder, onTextChange}) => {
  return (
    <View style={styles.container}>
      <TextInput
        placeholder={placeholder}
        style={styles.input}
        onChangeText={text => onTextChange(text)}
      />
      <TouchableOpacity style={styles.eyeIcon}>
        <Icon name={'search'} size={16} color={colors.primaryIconTintColor} />
      </TouchableOpacity>
    </View>
  );
};

export default SearchBar;
