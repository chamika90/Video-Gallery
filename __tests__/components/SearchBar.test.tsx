import React from 'react';
import 'react-native';
import {render} from '@testing-library/react-native';

import SearchBar from '../../src/components/SearchBar';

describe('SearchBar', () => {
  it('should match snapshot', async () => {
    const {toJSON} = render(
      <SearchBar
        placeholder={'search'}
        onTextChange={() => {
          console.log('on text changed');
        }}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should display text as search', async () => {
    const {getAllByPlaceholderText} = render(
      <SearchBar
        placeholder={'search'}
        onTextChange={() => {
          console.log('on text changed');
        }}
      />,
    );

    expect(getAllByPlaceholderText('search').length).toBe(1);
  });
});
