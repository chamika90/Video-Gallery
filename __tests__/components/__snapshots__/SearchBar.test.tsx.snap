// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 6,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 5,
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    placeholder="search"
    style={
      Object {
        "flex": 3,
      }
    }
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 0.5,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#9C9F9D",
            "fontSize": 16,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      ï€‚
    </Text>
  </View>
</View>
`;
